PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/rendering/sdk/UserAgentFetcherTask.java

```diff
@@ -0,0 +1,56 @@
+package org.prebid.mobile.rendering.sdk;
+
+import android.os.Build;
+import android.text.TextUtils;
+import android.webkit.WebSettings;
+
+import org.prebid.mobile.LogUtil;
+import org.prebid.mobile.rendering.utils.helpers.AppInfoManager;
+
+public class UserAgentFetcherTask implements Runnable {
+
+    private static final String TAG = "UserAgentFetcherTask";
+
+    @Override
+    public void run() {
+        String userAgent = "";
+
+        try {
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR1) {
+                userAgent = WebSettings.getDefaultUserAgent(PrebidContextHolder.getContext());
+            }
+        } catch (Exception any) {
+            LogUtil.error(TAG, "Failed to get user agent");
+        }
+
+        if (TextUtils.isEmpty(userAgent) || userAgent.contains("UNAVAILABLE")) {
+            userAgent = "Mozilla/5.0 (Linux; U; Android " + android.os.Build.VERSION.RELEASE + ";" + " " + getDeviceName() + ")";
+        }
+
+        AppInfoManager.setUserAgent(userAgent);
+    }
+
+
+    public static String getDeviceName() {
```

Comment:
Device specs are maybe not very correlated with a class that handles the user agent, a dedicated class to group functions to retrieve device specs could be good and better maintened

Reply:
Moved.